shimmingtoolbox/shim/sequencer.py

```diff
@@ -375,6 +377,10 @@ def select_optimizer(self):
                 optimizer = supported_optimizers[self.method](self.coils, self.nii_fieldmap.get_fdata(),
                                                               self.nii_fieldmap.affine, self.opt_criteria,
                                                               reg_factor=self.reg_factor)
+            elif self.method == 'gradient':
+                optimizer = supported_optimizers[self.method](self.coils, self.nii_fieldmap.get_fdata(),
+                                                              self.nii_fieldmap.affine, self.opt_criteria,
+                                                              reg_factor=self.reg_factor)
```

Comment:
This is the same code called for the "least_squares" method. You can simply do:
if self.method in ['least_squares', 'gradient'] :